ext/bcmath/libbcmath/src/recmul.c

```diff
@@ -205,17 +203,17 @@ static inline unsigned short bc_expand_lut(unsigned char c)
 
 /* Writes the character representation of the number encoded in value.
  * E.g. if value = 1234, then the string "1234" will be written to str. */
-static void bc_write_bcd_representation(uint32_t value, char *str)
+static void bc_write_bcd_representation(int32_t value, char *str)
 {
-	uint32_t upper = value / 100; /* e.g. 12 */
-	uint32_t lower = value % 100; /* e.g. 34 */
+	int32_t upper = value / 100; /* e.g. 12 */
+	int32_t lower = value % 100; /* e.g. 34 */
 
 #if BC_LITTLE_ENDIAN
 	/* Note: little endian, so `lower` comes before `upper`! */
-	uint32_t digits = bc_expand_lut(LUT[lower]) << 16 | bc_expand_lut(LUT[upper]);
+	int32_t digits = bc_expand_lut(LUT[lower]) << 16 | bc_expand_lut(LUT[upper]);
 #else
 	/* Note: big endian, so `upper` comes before `lower`! */
-	uint32_t digits = bc_expand_lut(LUT[upper]) << 16 | bc_expand_lut(LUT[lower]);
+	int32_t digits = bc_expand_lut(LUT[upper]) << 16 | bc_expand_lut(LUT[lower]);
 #endif
 	memcpy(str, &digits, sizeof(digits));
 }
```

Comment:
Arguably, this needs to take care of the sign of `value` or am I forgetting what this function is actually used for?

Reply:
In fact, negative values ​​are never passed to this function: if implement the Karatsuba algorithm, the values ​​are all adjusted to be positive before this function is called.

So, the type change for this function was not needed in the first place and I made an unnecessary change. Thanks.

Reply:
Reverted it